### ISSUE TITLE
NullPointerException in MetaImpl.snapshot() when foreign key column order doesn't match unique key column order

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11258]
The test table introduced in https://github.com/jOOQ/jOOQ/issues/8553 causes trouble also when calling `Meta.snapshot()`:

```sql
CREATE TABLE t_8553_a (
  c1 INT NOT NULL,
  c2 INT NOT NULL,
  
  CONSTRAINT pk_t_8553_a PRIMARY KEY (c1, c2)
);

CREATE TABLE t_8553_b (
  c1 INT NOT NULL,
  c2 INT NOT NULL,
  
  CONSTRAINT fk_t_8553_b FOREIGN KEY (c2, c1) REFERENCES t_8553_a (c2, c1)
);
```

There's an exception:

```
Exception in thread "main" java.lang.NullPointerException
	at org.jooq.impl.Snapshot$SnapshotTable.resolveReferences(Snapshot.java:261)
	at org.jooq.impl.Snapshot$SnapshotSchema.resolveReferences(Snapshot.java:152)
	at org.jooq.impl.Snapshot$SnapshotCatalog.resolveReferences(Snapshot.java:115)
	at org.jooq.impl.Snapshot.resolveReferences(Snapshot.java:87)
	at org.jooq.impl.Snapshot.<init>(Snapshot.java:82)
	at org.jooq.impl.AbstractMeta.snapshot(AbstractMeta.java:568)
```

----
See also: https://github.com/jOOQ/jOOQ/issues/11261

**Comment by lukaseder** [comment]
When running the test on an interpreted schema, we can't reproduce this because the interpreter "fixes" the column order (see #11261)

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.5 (#11264)
